docs/user/custom_layers.rst

```diff
@@ -0,0 +1,78 @@
+Creating custom layers
+======================
+
+A simple layer
+--------------
+
+To implement a custom layer in Lasagne, you will have to write a Python class
+that subclasses :class:`Layer` and implement at least one method:
+`get_output_for()`. This method computes the output of the layer given its
+input. Note that both the output and the input are Theano expressions, so they
+are symbolic.
+
+The following is an example implementation of a layer that multiplies its input
+by 2:
+
+.. code:: python
+
+    class DoubleLayer(lasagne.layers.Layer):
+        def get_output_for(self, input, **kwargs):
+            return 2 * input
+
+This is all that's required to implement a functioning custom layer class in
+Lasagne.
+
+A layer that changes the shape
+------------------------------
+
+If the layer does not change the shape of the data (for example because it
+applies an elementwise operation), then implementing only this one method is
+sufficient. Lasagne will assume that the output of the layer has the same shape
+as its input.
+
+However, if the operation performed by the layer changes the shape of the data,
+you should also implement `get_output_shape_for()`. This method computes the
+shape of the layer output given the shape of its input. Note that this shape
+computation should result in a tuple of integers, so it is *not* symbolic.
+
+This method exists because Lasagne needs a way to propagate shape information
+when a network is defined, so it can determine what sizes the parameter arrays
```

Comment:
"parameter array" sounds like it's referring to a list of parameters... maybe "weight matrices" or "parameter tensors"? (minor issue)